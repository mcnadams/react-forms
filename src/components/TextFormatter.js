import React, { PureComponent } from 'react';
import figlet from 'figlet';

export default class TextFormatter extends PureComponent {
  state = {
    input: '',
    formatted: '',
    color: ''
  }

  // formatInput = () => {
  //   const { input } = this.state;
  //   figlet.text(input, { font : 'Ghost' }, (err, result) => {
  //     console.log(err);
  //     this.setState({ formatted: result });
  //   });
  // }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    return (
      <>
        <input name='input' value={this.state.input} onChange={this.handleChange} />
        <input name='color' type='color' value={this.state.color} onChange={this.handleChange} />
        <div style={{ color: this.state.color }}>{this.state.input}</div>
      </>
    );
  }
}
