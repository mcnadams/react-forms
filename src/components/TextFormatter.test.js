import React from 'react'
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('Text Formatter', () => {
  it('text formatter', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
