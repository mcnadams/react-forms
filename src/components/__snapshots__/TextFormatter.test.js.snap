// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Formatter text formatter 1`] = `
<Fragment>
  <input
    name="input"
    onChange={[Function]}
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value=""
  />
  <div
    style={
      Object {
        "color": "",
      }
    }
  />
</Fragment>
`;
